crates/cli/src/parse/timestamps.rs

```diff
@@ -0,0 +1,462 @@
+            let parts: Vec<_> = second_ref.split('/').collect();
+            let (second_ref, n_keep) = if parts.len() == 2 {
+                let n_keep = parts[1].parse::<u32>().map_err(|_| {
+                    ParseError::ParseError("cannot parse timestamp interval size".to_string())
+                })?;
+                (parts[0], Some(n_keep))
+            } else {
+                (*second_ref, None)
+            };
+
+            let (start_timestamp, end_timestamp) =
+                parse_timestamp_range(first_ref, second_ref, fetcher).await?;
+            let (start_block, end_block) = (
+                timestamp_to_block_number(start_timestamp, fetcher).await?,
+                timestamp_to_block_number(end_timestamp, fetcher).await?,
+            );
+            block_range_to_block_chunk(start_block, end_block, as_range, None, n_keep)
+        }
+        _ => Err(ParseError::ParseError(
+            "timestamps must be in format timestamp or start_timestamp:end_timestamp".to_string(),
+        )),
+    }
+}
+
+async fn parse_timestamp_range<P>(
+    first_ref: &str,
+    second_ref: &str,
+    fetcher: &Fetcher<P>,
+) -> Result<(u64, u64), ParseError>
+where
+    P: JsonRpcClient,
+{
+    let (start_timestamp, end_timestamp) = match (first_ref, second_ref) {
+        _ if first_ref.starts_with('-') => {
+            let end_timestamp =
+                parse_timestamp_number(second_ref, RangePosition::Last, fetcher).await?;
+
+            let start_timestamp = end_timestamp
+                .checked_sub(
+                    parse_timestamp_number(&first_ref[1..], RangePosition::None, fetcher).await?,
+                )
+                .ok_or_else(|| ParseError::ParseError("start_timestamp underflow".to_string()))?;
+
+            (start_timestamp, end_timestamp)
+        }
+        _ if second_ref.starts_with('+') => {
+            let start_timestamp =
+                parse_timestamp_number(first_ref, RangePosition::First, fetcher).await?;
+
+            let end_timestamp = start_timestamp
+                .checked_add(
+                    parse_timestamp_number(&second_ref[1..], RangePosition::None, fetcher).await?,
+                )
+                .ok_or_else(|| ParseError::ParseError("end_timestamp overflow".to_string()))?;
+
+            (start_timestamp, end_timestamp)
+        }
+        _ => {
+            let start_timestamp =
+                parse_timestamp_number(first_ref, RangePosition::First, fetcher).await?;
+
+            let end_timestamp =
+                parse_timestamp_number(second_ref, RangePosition::Last, fetcher).await?;
+
+            (start_timestamp, end_timestamp)
+        }
+    };
+
+    let end_timestamp =
+        if second_ref != "latest" && !second_ref.is_empty() && !first_ref.starts_with('-') {
+            end_timestamp - 1
+        } else {
+            end_timestamp
+        };
+
+    Ok((start_timestamp, end_timestamp))
+}
+
+async fn parse_timestamp_number<P: JsonRpcClient>(
```

Comment:
instead of K M B units, should use time units

m = 60
h = 3600
d = 86400
w = 86400 * 7
M = 86400 * 30
y = 86400 * 365